R/generate_dust.R

```diff
@@ -540,16 +562,39 @@ generate_dust_core_data <- function(dat) {
 }
 
 
-odin_variable_transformer <- function(dat, rewrite) {
-  function(text, envir) {
+## It would be really nice to use glue for this but we can't disable
+## escaping whcih means that a '))' becomes ')' which results in
+## broken code. This approach is pretty ugly but should do the trick
+## for now.
+transform_compare_odin <- function(text, dat, rewrite) {
+  re <- "odin\\(\\s*([^) ]+)\\s*\\)"
+  i <- grep(re, text)
+  d <- gregexpr(re, text[i])
```

Comment:
What are `i` and `d`? The structure of text, and i and d derived from it are unclear to me here

Reply:
I've cleared that up now - let me know if that's easier. There's quite a bit of bookkeeping unfortunately

Reply:
Yep, that's a lot easier to follow!